README.md

```diff
@@ -548,6 +548,7 @@ For Kubernetes variants in AWS, the following settings are set for you automatic
 
 * `settings.kubernetes.cluster-dns-ip`: Derived from the EKS Service IP CIDR or the CIDR block of the primary network interface.
 * `settings.kubernetes.max-pods`: The maximum number of pods that can be scheduled on this node (limited by number of available IPv4 addresses)
+* `settings.kubernetes.hostname-override`: The node name kubelet uses as identification instead of the hostname or the name determined by the in-tree cloud provider if that's enabled. This *does not* override the `network.hostname` setting in any way, shape, or form. This is purely for configuring kubelet. **Note:** This is currently only automatically generated in `aws-k8s-1.26` variants and set to the private DNS name of the EC2 instance and left unset for all other K8s variants.
```

Comment:
```suggestion
* `settings.kubernetes.hostname-override`: The node name kubelet uses as identification instead of the hostname or the name determined by the in-tree cloud provider if that's enabled.

**Important note for all Kubernetes variants:** Changing this setting at runtime (not via user data) can cause issues with kubelet registration, as hostname is closely tied to the identity of the system for both registration and certificates/authorization purposes.

Most users don't need to change this setting. If left unset, the system hostname will be used instead. The `settings.network.hostname` setting can be used to specify the value for both `kubelet` and the host. Only set this override if you intend for the `kubelet` to register with a different name than the host.

For `aws-k8s-1.26` variants, which use the "external" cloud provider, a hostname override will be automatically generated by querying the EC2 API for the private DNS name of the instance. This is done for backwards compatibility with the deprecated "aws" cloud provider, which adjusted the hostname in a similar way. Future `aws-k8s-*` variants may remove this behavior.
```